api/src/main/java/com/epam/pipeline/manager/user/UserManager.java

```diff
@@ -83,10 +88,47 @@
     @Autowired
     private DataStorageValidator storageValidator;
 
-    @Transactional(propagation = Propagation.REQUIRED)
+    @Autowired
+    private FolderManager folderManager;
+
+    @Value("${storage.user.home.auto:false}")
+    private boolean shouldCreateDefaultHome;
+
+    @Value("${storage.user.home.template}")
+    private  String defaultUserStorageTemplateName;
+
     public PipelineUser createUser(String name, List<Long> roles,
                                    List<String> groups, Map<String, String> attributes,
                                    Long defaultStorageId) {
+        final PipelineUser newUser = createUser(name, roles, groups, attributes);
+        if (defaultStorageId != null) {
+            newUser.setDefaultStorageId(defaultStorageId);
+            userDao.updateUser(newUser);
+        } else if (shouldCreateDefaultHome) {
+            final AbstractDataStorage defaultStorage = createUserDefaultFolder(newUser).getStorages().get(0);
+            newUser.setDefaultStorageId(defaultStorage.getId());
+            userDao.updateUser(newUser);
+        }
+        return newUser;
+    }
+
+    private Folder createUserDefaultFolder(final PipelineUser user) {
+        final Folder folder = new Folder();
+        final String userName = user.getUserName();
+        folder.setName(userName);
+        try {
+            return folderManager.createFromTemplate(folder, defaultUserStorageTemplateName, false);
```

Comment:
after project is created we need to call `changeOwner` on all created entities (or it will be unavailable for user)

Reply:
We need to check, how this will work, when user is created from `SamlUserDetailsService` - listing templates may fail, if user is not authorized yet.

Reply:
Setting owner permissions manually